### ISSUE TITLE
Failed to load Optifine 1.14.4 HD U F4 pre2

### ISSUE THREAD

**Issue by refparo** [modmuss50/OptiFabric#58]
Fabric loader: `0.4.8+build.159`
Mods:
```
durabilityviewer-1.14-fabric0.2.7-1.5.jar
fabric-api-0.3.0+build.207.jar
fabricmod_VoxelMap-1.9.13_for_1.14.4.jar
optifabric-0.4.3.jar
preview_OptiFine_1.14.4_HD_U_F4_pre2.jar
RoughlyEnoughItems-2.9.7+build.143.jar
```

Everything works fine with Optifine HD U F3.

**Comment by refparo** [comment]
<details><summary>Log</summary>

```log
[22:12:10] [main/INFO]: Loading for game Minecraft 1.14.4
[22:12:15] [main/WARN]: Warnings were found! 
 - Mod gbfabrictools recommends mod {modmenu @ [*]}, which is missing!
 - Mod roughlyenoughitems recommends mod {cloth-config @ [>=0.1.0]}, which is missing!
 - Mod durabilityviewer recommends mod {modmenu @ [*]}, which is missing!
[22:12:15] [main/WARN]: Mod `cloth-config2` (0.2.0) uses 'requires' key in fabric.mod.json, which is not supported - use 'depends'
[22:12:15] [main/WARN]: Mod `mm` (1.6) uses 'requires' key in fabric.mod.json, which is not supported - use 'depends'
[22:12:15] [main/WARN]: Mod `optifabric` (0.4.3) uses 'requires' key in fabric.mod.json, which is not supported - use 'depends'
[22:12:15] [main/WARN]: Mod `voxelmap` (1.9.13) uses 'requires' key in fabric.mod.json, which is not supported - use 'depends'
[22:12:15] [main/WARN]: Mod `cloth` (0.3.1+build.23) uses 'requires' key in fabric.mod.json, which is not supported - use 'depends'
[22:12:15] [main/INFO]: [FabricLoader] Loading 40 mods: fabric-renderer-api-v1@0.1.1+9b03381f42, fabric-networking-blockentity-v0@0.1.1+b494ebeb42, fabric-keybindings-v0@0.1.1+7dfef87c42, gbfabrictools@1.0.0, fabricloader@0.4.8+build.159, org_zeroturnaround_zt-zip@1.13, fabric-renderer-indigo@0.1.9+9b03381f42, fabric-containers-v0@0.1.2+b494ebeb42, fabric-biomes-v1@0.1.0+896c7fbb42, roughlyenoughitems@2.9.7+build.143, org_slf4j_slf4j-simple@1.7.26, fabric-crash-report-info-v1@0.1.0+5914746342, fabric-events-interaction-v0@0.1.0+ea100f6142, net_fabricmc_stitch@0.2.1.61, fabric-api-base@0.1.0+b494ebeb42, fabric-rendering-v0@0.1.1+6dad974e42, cloth-config2@0.2.0, mm@1.6, fabric-rendering-data-attachment-v1@0.1.0+b494ebeb42, fabric-resource-loader-v0@0.1.1+7495ea2842, fabric-textures-v0@0.1.4+b494ebeb42, fabric-content-registries-v0@0.1.1+b494ebeb42, fabric-tag-extensions-v0@0.1.1+b494ebeb42, blue_endless_jankson@1.1.0, optifabric@0.4.3, fabric-rendering-fluids-v1@0.1.0+dc4c57c242, fabric-commands-v0@0.1.1+b494ebeb42, fabric-registry-sync-v0@0.2.2+9b03381f42, durabilityviewer@1.14-fabric0.2.7-1.5, fabric-mining-levels-v0@0.1.0+ab421b9c42, voxelmap@1.9.13, fabric-events-lifecycle-v0@0.1.0+5914746342, fabric-loot-tables-v1@0.1.0+b494ebeb42, fabric@0.3.0+build.207, org_slf4j_slf4j-api@1.7.26, fabric-item-groups-v0@0.1.0+1e69a0a542, fabric-models-v0@0.1.0+5914746342, fabric-object-builders-v0@0.1.1+9b03381f42, fabric-networking-v0@0.1.2+9b03381f42, cloth@0.3.1+build.23
[22:12:15] [main/WARN]: Mod `net_fabricmc_stitch` (0.2.1.61) uses more dot-separated version components than SemVer allows; support for this is currently not guaranteed.
[22:12:15] [main/INFO]: SpongePowered MIXIN Subsystem Version=0.7.11 Source=file:/mnt/Documents/HMCL/libraries/net/fabricmc/sponge-mixin/0.7.11.38/sponge-mixin-0.7.11.38.jar Service=Knot/Fabric Env=CLIENT
[22:12:15] [main/INFO]: Compatibility level set to JAVA_8
[22:12:17] [main/ERROR]: Error encountered during mixin config postInit step'mixins.mm.json': Failed to setup optifine
java.lang.RuntimeException: Failed to setup optifine
	at me.modmuss50.optifabric.mod.OptifabricSetup.run(OptifabricSetup.java:37) ~[optifabric-0.4.3.jar:?]
	at com.chocohead.mm.Plugin.getMixins(Plugin.java:289) ~[28d81515-4c6d-47cd-ab3c-be4dd8b81a42.jar:?]
	at org.spongepowered.asm.mixin.transformer.MixinConfig.postInitialise(MixinConfig.java:524) ~[sponge-mixin-0.7.11.38.jar:0.7.11.38+unknown-b0.git-unknown]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.prepareConfigs(MixinTransformer.java:675) [sponge-mixin-0.7.11.38.jar:0.7.11.38+unknown-b0.git-unknown]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.select(MixinTransformer.java:572) [sponge-mixin-0.7.11.38.jar:0.7.11.38+unknown-b0.git-unknown]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.checkSelect(MixinTransformer.java:548) [sponge-mixin-0.7.11.38.jar:0.7.11.38+unknown-b0.git-unknown]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:451) [sponge-mixin-0.7.11.38.jar:0.7.11.38+unknown-b0.git-unknown]
	at net.fabricmc.loader.launch.knot.KnotClassDelegate.loadClassData(KnotClassDelegate.java:179) [fabric-loader-0.4.8+build.159.jar:?]
	at net.fabricmc.loader.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:142) [fabric-loader-0.4.8+build.159.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357) [?:1.8.0_222]
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:168) [fabric-loader-0.4.8+build.159.jar:?]
	at net.fabricmc.loader.launch.knot.Knot.init(Knot.java:129) [fabric-loader-0.4.8+build.159.jar:?]
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:26) [fabric-loader-0.4.8+build.159.jar:?]
Caused by: java.lang.NullPointerException
	at java.util.jar.JarFile.getInputStream(JarFile.java:463) ~[?:1.8.0_222]
	at me.modmuss50.optifabric.patcher.ASMUtils.asClassNode(ASMUtils.java:30) ~[optifabric-0.4.3.jar:?]
	at me.modmuss50.optifabric.patcher.LambadaRebuiler.buildClassMap(LambadaRebuiler.java:57) ~[optifabric-0.4.3.jar:?]
	at me.modmuss50.optifabric.patcher.LambadaRebuiler.buildLambadaMap(LambadaRebuiler.java:38) ~[optifabric-0.4.3.jar:?]
	at me.modmuss50.optifabric.mod.OptifineSetup.getRuntime(OptifineSetup.java:85) ~[optifabric-0.4.3.jar:?]
	at me.modmuss50.optifabric.mod.OptifabricSetup.run(OptifabricSetup.java:25) ~[optifabric-0.4.3.jar:?]
	... 12 more
[22:12:17] [main/INFO]: Setting user: problem233
[22:12:20] [main/INFO]: Definitely not up to no good
[22:12:20] [main/INFO]: [Indigo] Registering Indigo renderer!
[22:12:20] [main/INFO]: [Indigo] Compatibility mode enabled.
[22:12:21] [main/INFO]: [REI] Config is loaded.
[22:12:21] [main/INFO]: [REI] Registered plugin roughlyenoughitems:default_plugin from DefaultPlugin
[22:12:21] [main/WARN]: Skipping bad option: lastServer:
[22:12:21] [main/INFO]: LWJGL Version: 3.2.2 build 10
[22:12:23] [main/WARN]: ERROR : Couldn't load Narrator library : Unable to load library 'fliteWrapper': Native library (linux-x86-64/libfliteWrapper.so) not found in resource path (/mnt/Documents/HMCL/libraries/net/fabricmc/tiny-mappings-parser/0.1.1.8/tiny-mappings-parser-0.1.1.8.jar:/mnt/Documents/HMCL/libraries/net/fabricmc/sponge-mixin/0.7.11.38/sponge-mixin-0.7.11.38.jar:/mnt/Documents/HMCL/libraries/net/fabricmc/tiny-remapper/0.1.0.33/tiny-remapper-0.1.0.33.jar:/mnt/Documents/HMCL/libraries/net/fabricmc/fabric-loader-sat4j/2.3.5.4/fabric-loader-sat4j-2.3.5.4.jar:/mnt/Documents/HMCL/libraries/com/google/jimfs/jimfs/1.1/jimfs-1.1.jar:/mnt/Documents/HMCL/libraries/org/ow2/asm/asm/7.1/asm-7.1.jar:/mnt/Documents/HMCL/libraries/org/ow2/asm/asm-analysis/7.1/asm-analysis-7.1.jar:/mnt/Documents/HMCL/libraries/org/ow2/asm/asm-commons/7.1/asm-commons-7.1.jar:/mnt/Documents/HMCL/libraries/org/ow2/asm/asm-tree/7.1/asm-tree-7.1.jar:/mnt/Documents/HMCL/libraries/org/ow2/asm/asm-util/7.1/asm-util-7.1.jar:/mnt/Documents/HMCL/libraries/com/google/guava/guava/21.0/guava-21.0.jar:/mnt/Documents/HMCL/libraries/net/fabricmc/intermediary/1.14.4/intermediary-1.14.4.jar:/mnt/Documents/HMCL/libraries/net/fabricmc/fabric-loader/0.4.8+build.159/fabric-loader-0.4.8+build.159.jar:/mnt/Documents/HMCL/libraries/com/mojang/patchy/1.1/patchy-1.1.jar:/mnt/Documents/HMCL/libraries/oshi-project/oshi-core/1.1/oshi-core-1.1.jar:/mnt/Documents/HMCL/libraries/net/java/dev/jna/jna/4.4.0/jna-4.4.0.jar:/mnt/Documents/HMCL/libraries/net/java/dev/jna/platform/3.4.0/platform-3.4.0.jar:/mnt/Documents/HMCL/libraries/com/ibm/icu/icu4j-core-mojang/51.2/icu4j-core-mojang-51.2.jar:/mnt/Documents/HMCL/libraries/com/mojang/javabridge/1.0.22/javabridge-1.0.22.jar:/mnt/Documents/HMCL/libraries/net/sf/jopt-simple/jopt-simple/5.0.3/jopt-simple-5.0.3.jar:/mnt/Documents/HMCL/libraries/io/netty/netty-all/4.1.25.Final/netty-all-4.1.25.Final.jar:/mnt/Documents/HMCL/libraries/com/google/guava/guava/21.0/guava-21.0.jar:/mnt/Documents/HMCL/libraries/org/apache/commons/commons-lang3/3.5/commons-lang3-3.5.jar:/mnt/Documents/HMCL/libraries/commons-io/commons-io/2.5/commons-io-2.5.jar:/mnt/Documents/HMCL/libraries/commons-codec/commons-codec/1.10/commons-codec-1.10.jar:/mnt/Documents/HMCL/libraries/net/java/jinput/jinput/2.0.5/jinput-2.0.5.jar:/mnt/Documents/HMCL/libraries/net/java/jutils/jutils/1.0.0/jutils-1.0.0.jar:/mnt/Documents/HMCL/libraries/com/mojang/brigadier/1.0.17/brigadier-1.0.17.jar:/mnt/Documents/HMCL/libraries/com/mojang/datafixerupper/2.0.24/datafixerupper-2.0.24.jar:/mnt/Documents/HMCL/libraries/com/google/code/gson/gson/2.8.0/gson-2.8.0.jar:/mnt/Documents/HMCL/libraries/com/mojang/authlib/1.5.25/authlib-1.5.25.jar:/mnt/Documents/HMCL/libraries/org/apache/commons/commons-compress/1.8.1/commons-compress-1.8.1.jar:/mnt/Documents/HMCL/libraries/org/apache/httpcomponents/httpclient/4.3.3/httpclient-4.3.3.jar:/mnt/Documents/HMCL/libraries/commons-logging/commons-logging/1.1.3/commons-logging-1.1.3.jar:/mnt/Documents/HMCL/libraries/org/apache/httpcomponents/httpcore/4.3.2/httpcore-4.3.2.jar:/mnt/Documents/HMCL/libraries/it/unimi/dsi/fastutil/8.2.1/fastutil-8.2.1.jar:/mnt/Documents/HMCL/libraries/org/apache/logging/log4j/log4j-api/2.8.1/log4j-api-2.8.1.jar:/mnt/Documents/HMCL/libraries/org/apache/logging/log4j/log4j-core/2.8.1/log4j-core-2.8.1.jar:/mnt/Documents/HMCL/libraries/org/lwjgl/lwjgl/3.2.2/lwjgl-3.2.2.jar:/mnt/Documents/HMCL/libraries/org/lwjgl/lwjgl-jemalloc/3.2.2/lwjgl-jemalloc-3.2.2.jar:/mnt/Documents/HMCL/libraries/org/lwjgl/lwjgl-openal/3.2.2/lwjgl-openal-3.2.2.jar:/mnt/Documents/HMCL/libraries/org/lwjgl/lwjgl-opengl/3.2.2/lwjgl-opengl-3.2.2.jar:/mnt/Documents/HMCL/libraries/org/lwjgl/lwjgl-glfw/3.2.2/lwjgl-glfw-3.2.2.jar:/mnt/Documents/HMCL/libraries/org/lwjgl/lwjgl-stb/3.2.2/lwjgl-stb-3.2.2.jar:/mnt/Documents/HMCL/libraries/com/mojang/text2speech/1.11.3/text2speech-1.11.3.jar:/mnt/Documents/HMCL/versions/Vanilla 1.14.4/Vanilla 1.14.4.jar)
[22:13:29] [main/INFO]: OpenAL initialized.
[22:13:29] [main/INFO]: Sound engine started
[22:13:29] [main/INFO]: Created: 1024x512 textures-atlas
[22:13:30] [main/INFO]: Created: 256x256 textures/particle-atlas
[22:13:30] [main/INFO]: Created: 256x256 textures/painting-atlas
[22:13:30] [main/INFO]: Created: 256x128 textures/mob_effect-atlas
[22:13:30] [main/INFO]: [STDOUT]: CATEGORY ORDER IS 9
[22:13:30] [main/INFO]: Created: 256x256 waypoints-atlas
[22:13:30] [main/INFO]: Created: 256x128 chooser-atlas
[22:13:30] [main/INFO]: Created: 512x256 mobs-atlas
[22:13:30] [main/INFO]: Created: 64x64 pings-atlas
[22:13:39] [main/INFO]: Stopping!
```

</details>

**Comment by Dinolek** [comment]
Starting from F4 pre1, optifine now have forge support, so srg's are included.
They need to be blacklisted from LambadaRebuiler and optifinePatches creation.

**Comment by modmuss50** [comment]
if optifine is in srg format, I wont be able to support it due to leagal reasons. I will look into this tomorrow, lets hope thats not the case.

**Comment by modmuss50** [comment]
Ok, I took some time to look into this, and the good news is optifine includes srg and none srg classes! I am testing a version that removes the srg classes right now.

**Comment by modmuss50** [comment]
I have released optifabric-0.5.1.jar that should fix this issue. Please test (You may need to delete the .optifabric folder if you tryied with the old version)

If you still have any issues please open a NEW issue. Thanks.
